### PR TITLE
Modify to allow datatype return

### PR DESCRIPTION
**Pull request by astrovsky01**

### CODE REVIEW

File: galaxy_ie_helpers/__init__.py [hunk 1]
```diff
@@ -186,17 +187,24 @@ def get(datasets_identifiers, identifier_type='hid', history_id=None):
             dc = DatasetClient(gi)
             history = hc.show_history(history_id, contents=True)
             datasets = {ds[identifier_type]: ds['id'] for ds in history}
+            if retrieve_datatype:
+                datatypes = {{ds[identifier_type]: ds['extension'] for ds in history}}
```
Comment (bgruening) [inline galaxy_ie_helpers/__init__.py]: why do you need here `{{`?

File: galaxy_ie_helpers/__init__.py [hunk 2]
```diff
@@ -186,17 +187,24 @@ def get(datasets_identifiers, identifier_type='hid', history_id=None):
...
             log.debug('Cached, not re-downloading')
 
         file_path_all.append(file_path)
+        if retrieve_datatype:
+            datatypes_all.append(datatypes)
 
     ## First path if only one item given, otherwise all paths.
     ## Should not break compatibility.
-    return file_path_all[0] if len(file_path_all) == 1 else file_path_all
+    if retrieve_datatype == True:
```
Comment (bgruening) [inline galaxy_ie_helpers/__init__.py]: == True can be removed

File: galaxy_ie_helpers/__init__.py [hunk 3]
```diff
@@ -186,17 +187,24 @@ def get(datasets_identifiers, identifier_type='hid', history_id=None):
...
+        return file_path_all[0] if len(file_path_all) == 1 else file_path_all, datatypes_all[0] if len(file_path_all) == 1 else datatypes_all
```
Comment (bgruening) [inline galaxy_ie_helpers/__init__.py]: I guess this should be written in multiple lines for enhancing readability.

File: galaxy_ie_helpers/__init__.py (hunk 1 above)
Comment (astrovsky01) [reply galaxy_ie_helpers/__init__.py]: Copy/paste-based remnant.

File: galaxy_ie_helpers/__init__.py [hunk 4]
```diff
@@ -201,8 +201,9 @@ def get(datasets_identifiers, identifier_type='hid', history_id=None, retrieve_d
 
     ## First path if only one item given, otherwise all paths.
     ## Should not break compatibility.
-    if retrieve_datatype == True:
-        return file_path_all[0] if len(file_path_all) == 1 else file_path_all, datatypes_all[0] if len(file_path_all) == 1 else datatypes_all
+    if retrieve_datatype:
+        return file_path_all[0] if len(file_path_all) == 1  \
```
Comment (bgruening) [inline galaxy_ie_helpers/__init__.py]: What I mean is something like:
```
 if len(file_path_all) == 1 else datatypes_all:
    if len(file_path_all) == 1 
        return file_path_all[0] 
    else:
           return file_path_all, datatypes_all[0]
```
 
probably wrong, from my cellphone

**Comment by bgruening** [comment]
LGTM.

**Comment by bgruening** [comment]
Still draft?

**Comment by astrovsky01** [comment]
Trying to figure out how to test it without deploying -- needs to run on a server. If you think it's good, I trust you, but wanted to figure out how to run it first, personally

**Comment by bgruening** [comment]
You can install it in the running Docker container with `pip install --user git+https://github.com/d1b/pip.git@fix_pip_build_directory`

**Comment by astrovsky01** [comment]
Right, but It needs to run on a galaxy IT/IE to be relevant. Can I do that on EU's IT?

**Comment by bgruening** [comment]
Sure. I think this should work.

**Comment by astrovsky01** [comment]
Sorry for all the emails -- the cached file prevented datatype from getting retrieved

**Comment by astrovsky01** [comment]
Got a single dataset working, now to fix multiple

**Comment by astrovsky01** [comment]
@bgruening Done

**Review by bgruening — APPROVED** [review]

**Pull request merged**